### ISSUE TITLE
Inventory Weapons - Assuming Aggressive for Controlled Weapons

### ISSUE THREAD

**Issue by whartd** [Matt12yearslater/max-hit#5]
The code for calculating max hits for inventory weapons assumes that the attack style is aggressive even for weapons without an aggressive option, leading to incorrect values. After reading the code I thought this might be the case, and I have tested it and confirmed it with a dragon hunter lance.  I'm not sure how to fix this. Might it be possible somewhere to find the available attack styles for each inventory weapon and choose the one with the highest str bonus? I.e. aggr>controlled>nothing

**Comment by Matt12yearslater** [comment]
Yeah I didn’t know how to get around that, the way I’m getting attack styles from weapons is seeing what options are available once equipped, I don’t know how to find the attack styles when the weapon is unequipped, but if someone can find that itd definitely be helpful.

**Comment by Matt12yearslater** [comment]
It would also be really cool to have it remember which style was selected for that weapon, like if you’ve selected rapid on knives and accurate on a bow, but again I’m not super sure how to implement that.

**Comment by whartd** [comment]
yeah, I did some poking around after I posted that and nothing was popping out at me as a solution for finding possible attack styles for unequipped weapons. And hmm, I'm not sure how to implement that either. I could picture something like the custom shift click or inventory tags where you right click your inventory icon to enable right click options, then can right click each weapon in your inventory and choose an attack style for it, but I think that'd be overboard for this kind of plugin. 

Does accurate actually have a higher max hit for range? I never looked at that part of the formula.

**Comment by Matt12yearslater** [comment]
Haha maybe, it could just be an option in the config menu but I didn’t add that because I didn’t think the clutter was worth it. Could be though, I’m not the ultimate decider. 

And yeah accurate gives a style bonus similar to aggressive for range, it’s really noticeable if you test with ballista or something.

**Comment by Seebity** [comment]
Closing as not planned/intended
